### PR TITLE
Add AWS provider specific config to the shoot converter

### PR DESCRIPTION
**Pull request by akgalwas**
**Description**

Changes proposed in this pull request:

- AWS provider specific config implemented

### CODE REVIEW

File: internal/gardener/shoot/hyperscaler/aws/provider_specific_config.go [hunk 1]
```diff
@@ -2,12 +2,43 @@ package aws
...
+	}
+}
+
+func NewControlPlaneConfig() *ControlPlaneConfig {
+	return &ControlPlaneConfig{
+		TypeMeta: v1.TypeMeta{
+			Kind:       controlPlaneConfigKind,
+			APIVersion: apiVersion,
+		},
+	}
```
Comment (koala7659) [inline internal/gardener/shoot/hyperscaler/aws/provider_specific_config.go:43]: We are missing AWS WorkerConfig configuration. See the function [func (c AWSGardenerConfig) ExtendShootConfig](https://github.com/kyma-project/control-plane/blob/ea75326a0e2f1108f4297dd04b8090cd75266898/components/provisioner/internal/model/gardener_config.go#L632) in Provisioner repository. 


There are three elements configured: 

1. InfrastructureConfig
2. ControlPlaneConfig
3. Workers with AWS specific provider config taking into account parameter `EnableIMDSv2` [added recently](https://github.com/kyma-project/control-plane/pull/3429) (I'm not sure if could be missing in Runtime CR)

Code from provisioner with AWS specific worker configuration: 

```golang
	if c.input.EnableIMDSv2 != nil && *c.input.EnableIMDSv2 {
		awsWorkerConfig := NewAWSWorkerConfig(awsIMDSv2HTTPPutResponseHopLimit)
		jsonWCData, err := json.Marshal(awsWorkerConfig)
		if err != nil {
			return apperrors.Internal("error encoding aws worker config: %s", err.Error())
		}
		workers[0].ProviderConfig = &apimachineryRuntime.RawExtension{Raw: jsonWCData}
	}
```

**Review by koala7659 — CHANGES_REQUESTED** [review]
Please check my comments

File: internal/gardener/shoot/hyperscaler/aws/provider_specific_config_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	t.Run("create Control Plane config", func(t *testing.T) {
+		// when
+		controlPlaneConfigBytes, err := GetControlPlaneConfig()
+
+		// then
+		require.NoError(t, err)
+
+		var controlPlaneConfig ControlPlaneConfig
+		err = json.Unmarshal(controlPlaneConfigBytes, &controlPlaneConfig)
+		assert.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply internal/gardener/shoot/hyperscaler/aws/provider_specific_config_test.go:21]: This is done on purpose. Structure of the objects, and unmarshalling is tested in provider_specific_config_test.go

File: internal/gardener/shoot/hyperscaler/aws/model.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package aws
+
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply internal/gardener/shoot/hyperscaler/aws/model.go:2]: This is copied from the Provisioner. After quick look I didn't find where the types are defined. I will double check that.

File: internal/gardener/shoot/hyperscaler/aws/provider_specific_config_test.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		"Regular 10.250.0.0/16": {
+			givenNodesCidr: "10.250.0.0/16",
+			givenZoneNames: []string{
+				"eu-central-1a",
+				"eu-central-1b",
+				"eu-central-1c",
+			},
+			expectedAwsZones: []Zone{
+				{
+					Workers:  "10.250.0.0/19",
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply internal/gardener/shoot/hyperscaler/aws/provider_specific_config_test.go:40]: Done

File: internal/gardener/shoot/hyperscaler/aws/provider_specific_config_test.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+			// when
+			var infrastructureConfig InfrastructureConfig
+			err = json.Unmarshal(infrastructureConfigBytes, &infrastructureConfig)
+
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, apiVersion, infrastructureConfig.TypeMeta.APIVersion)
+			assert.Equal(t, infrastructureConfigKind, infrastructureConfig.TypeMeta.Kind)
+
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply internal/gardener/shoot/hyperscaler/aws/provider_specific_config_test.go:97]: Done

File: internal/gardener/shoot/hyperscaler/aws/model.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// This types are copied from https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-azure/pkg/apis/azure/types_infrastructure.go as it does not contain json tags
+
```
[the remark this answers is not in the document]
Comment (akgalwas) [reply internal/gardener/shoot/hyperscaler/aws/model.go:8]: I changed the comment as it is misleading.

File: internal/gardener/shoot/hyperscaler/aws/provider_specific_config.go (hunk 1 above)
Comment (akgalwas) [reply internal/gardener/shoot/hyperscaler/aws/provider_specific_config.go:50]: Done

**Review by koala7659 — APPROVED** [review]
LGTM

**Pull request merged**